stripe-ui-core/src/main/java/com/stripe/android/uicore/analytics/UiEventReporterCompose.kt

```diff
@@ -0,0 +1,81 @@
+package com.stripe.android.uicore.analytics
+
+import android.os.Parcelable
+import androidx.annotation.RestrictTo
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.PressInteraction
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.compositionLocalOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.SaverScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import com.stripe.android.core.Logger
+import com.stripe.android.uicore.BuildConfig
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.parcelize.Parcelize
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+val LocalUiEventReporter = compositionLocalOf<UiEventReporter> {
+    EmptyUiEventReporter
+}
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+@Composable
+fun rememberInteractionReporter(): Pair<MutableInteractionSource, () -> Unit> {
```

Comment:
I think it's worth creating a type to return. It's a bit distracting to figure out the magic that's happening here. Would be easier to read by just referencing properties directly.